src/injector/inject.ts

```diff
@@ -397,24 +401,28 @@ export function inject (vm: ChibiCompatibleVM) {
                 // Add temporarily load from file button
                 const sideloadTempButton = document.createElement('button');
                 sideloadTempButton.setAttribute('text', format('chibi.sideloadTemporarily'));
-                sideloadTempButton.setAttribute('callbackKey', 'CHIBI_SIDELOAD_FROM_FILE_TEMPORAILY');
+                sideloadTempButton.setAttribute(
+                    'callbackKey',
+                    'CHIBI_SIDELOAD_FROM_FILE_TEMPORAILY'
+                );
                 workspace.registerButtonCallback('CHIBI_SIDELOAD_FROM_FILE_TEMPORAILY', () => {
                     const input = document.createElement('input');
                     input.setAttribute('type', 'file');
                     input.setAttribute('accept', '.js');
                     input.setAttribute('multiple', 'true');
-                    input.onchange = async (event: Event) => {
+                    input.addEventListener('change', async (event: Event) => {
                         const files = (event.target as HTMLInputElement).files;
                         if (!files) return;
                         for (const file of files) {
-                            const fileName = file.name;
-
-                            const url = URL.createObjectURL(file);
+                            const url = `data:text/javascript;charset=utf-8,${encodeURIComponent(
```

Comment:
This feature is intended to facilitate debugging extensions rather than placing extensions directly into project files, which will bring security. use ``createObjectURL`` instead.

Reply:
also, use revokeObjectURL if needed